torch/functional.py

```diff
@@ -361,7 +371,17 @@ def parse_subscript(n: int) -> str:
         # in the original implementation this line is omitted
         return einsum(equation, *_operands)
 
-    return _VF.einsum(equation, operands)  # type: ignore[attr-defined]
+    if len(operands) <= 2:
+        # the optimal path for contracting 0 or 1 time(s) is already optimized
+        return _VF.einsum(equation, operands)  # type: ignore[attr-defined]
+
+    flattened_path = None
+    if HAS_OPT_EINSUM:
+        path = contract_path(equation, *operands)[0]
```

Comment:
If it is significant overhead to compute the path, we may want to cache results here

Reply:
would this be by using lru_cache from functools? I did a brief search and that code's not used throughout torch so I'm wondering if there's another caching mechanism I should know about!

Reply:
We use `lru_cache` here and there.. but we should avoid caching any tensors...

Reply:
Yea..it causes cuda mem leaks so I ended up not doing it.